packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/cli.integtest.ts

```diff
@@ -2252,6 +2253,87 @@ integTest(
   }),
 );
 
+/**
+ * Create an S3 bucket, orphan that bucket, then import the bucket, with a NodeJSFunction lambda also in the stack.
+ *
+ * Validates fix for https://github.com/aws/aws-cdk/issues/31999 (import fails)
+ */
+integTest(
+  'test s3 bucket import with nodejsfunction lambda',
+  withDefaultFixture(async (fixture) => {
+    // Create a temporary directory for esbuild installation
+    const tempDir = mkdtempSync(path.join(os.tmpdir(), 'esbuild-'));
+    const originalPath = process.env.PATH;
+
+    // Install esbuild locally in the temporary directory
+    execSync('npm init -y && npm install esbuild', { cwd: tempDir, stdio: 'inherit' });
```

Comment:
We need to install `esbuild` to bundle the `NodeJSFunction` code. We install it in a temp directory, which is added to PATH.

At the end of the test, the temp directory is removed, and PATH is set back to its original value.

Reply:
But `esbuild` is already installed after you run `yarn install` - there should be no need to explicitly install it during the test. There must already be a test that defines a `NodeJsFunction` - does it do the same?

Reply:
I did find another [NodeJsFunction](https://github.com/aws/aws-cdk/blob/a7d73732cc41db518bd92eac3fd2e0f159ab3b7a/packages/%40aws-cdk-testing/cli-integ/resources/cdk-apps/sam_cdk_integ_app/lib/test-stack.js#L92-L99), it passes the option `forceDockerBundling: true`, which doesn't require `esbuild` as a runtime dependency. I've added this option to my test.